arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,199 @@
+        newFiles: [],
+        deletedFiles: [],
+    };
+
+    if (props.initialValue) {
+        currentValues.value = props.initialValue;
+    }
+});
+
+const currentMax = computed(() => {
+    if (currentValues.value) {
+        return (
+            props.nodeData.config.maxFiles - (currentValues.value.length ?? 0)
+        );
+    } else {
+        return props.nodeData.config.maxFiles;
+    }
+});
+
+function resolver(e: FormFieldResolverOptions) {
+    validate(e);
+}
+
+function select(event: FileUploadSelectEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
+
+function remove(event: FileUploadRemoveEvent) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        newFiles: event.files,
+    };
+}
+
+function validate(e: FormFieldResolverOptions) {
+    console.log("validate", e);
+}
+
+function deleteImage(fileId: string) {
+    // @ts-expect-error - This is a bug in the PrimeVue types
+    formFieldRef.value!.field.states.value = {
+        // @ts-expect-error - This is a bug in the PrimeVue types
+        ...formFieldRef.value!.field.states.value,
+        deletedFiles: [
+            // @ts-expect-error - This is a bug in the PrimeVue types
+            ...formFieldRef.value!.field.states.value.deletedFiles,
+            fileId,
+        ],
+    };
+    if (currentValues.value) {
+        const fileIndex = currentValues.value
+            .map((image) => image.file_id)
+            .indexOf(fileId);
+
+        if (fileIndex != -1) {
+            currentValues.value?.splice(fileIndex, 1);
+        }
+    }
+}
+</script>
+
+<template>
+    <FormField
+        ref="formFieldRef"
+        v-slot="$field"
+        :name="props.nodeAlias"
+        :initial-value="props.initialValue"
+        :resolver="resolver"
+    >
+        <div class="uploadedImagesContainer">
+            <div
+                v-for="(image, index) in currentValues"
+                :key="index"
+                class="uploadedImageRow"
+            >
+                <Image
+                    :key="index"
+                    class="uploadedImage"
+                    :src="image.url"
+                    :alt="image.name"
+                ></Image>
+                <div class="spacer"></div>
+                <Button
+                    icon="pi pi-trash"
+                    :aria-label="$gettext('delete')"
+                    severity="danger"
+                    outlined
+                    @click="deleteImage(image.file_id)"
+                />
+            </div>
+        </div>
+        currentMax: {{ currentMax }}
```

Comment:
needs <span> wrapping

Reply:
This was already yanked in an unpushed change - sorry about that.